core/src/layers/madsim.rs

```diff
@@ -0,0 +1,316 @@
+        unimplemented!("currently not supported, will be implemented in the future")
+    }
+
+    fn poll_next(&mut self, cx: &mut Context<'_>) -> Poll<Option<crate::Result<Bytes>>> {
+        unimplemented!("currently not supported, will be implemented in the future")
+    }
+}
+
+impl oio::BlockingRead for MadsimReader {
+    fn read(&mut self, buf: &mut [u8]) -> crate::Result<usize> {
+        unimplemented!("currently not supported, will be implemented in the future")
+    }
+
+    fn seek(&mut self, pos: SeekFrom) -> crate::Result<u64> {
+        unimplemented!("currently not supported, will be implemented in the future")
+    }
+
+    fn next(&mut self) -> Option<crate::Result<Bytes>> {
+        unimplemented!("currently not supported, will be implemented in the future")
+    }
+}
+
+pub struct MadsimWriter {
+    path: String,
+    args: OpWrite,
+    handle: ServiceNodeHandle,
+}
+
+impl oio::BlockingWrite for MadsimWriter {
+    fn write(&mut self, bs: Bytes) -> crate::Result<()> {
+        unimplemented!("currently not supported, will be implemented in the future")
+    }
+
+    fn append(&mut self, bs: Bytes) -> crate::Result<()> {
+        unimplemented!("currently not supported, will be implemented in the future")
+    }
+
+    fn close(&mut self) -> crate::Result<()> {
+        unimplemented!("currently not supported, will be implemented in the future")
+    }
+}
+
+#[async_trait]
+impl oio::Write for MadsimWriter {
+    async fn write(&mut self, bs: Bytes) -> crate::Result<()> {
+        self.handle.write(&self.path, &self.args, bs);
+        Ok(())
+    }
+
+    async fn append(&mut self, bs: Bytes) -> crate::Result<()> {
+        unimplemented!("currently not supported, will be implemented in the future")
+    }
+
+    async fn abort(&mut self) -> crate::Result<()> {
+        unimplemented!("currently not supported, will be implemented in the future")
+    }
+
+    async fn close(&mut self) -> crate::Result<()> {
+        Ok(())
+    }
+}
+
+pub struct MadsimPager {}
+
+#[async_trait]
+impl oio::Page for MadsimPager {
+    async fn next(&mut self) -> crate::Result<Option<Vec<Entry>>> {
+        unimplemented!("currently not supported, will be implemented in the future")
+    }
+}
+
+impl oio::BlockingPage for MadsimPager {
+    fn next(&mut self) -> crate::Result<Option<Vec<Entry>>> {
+        unimplemented!("currently not supported, will be implemented in the future")
+    }
+}
+
+#[derive(Default)]
+pub struct ServiceSim {
+    handles: Mutex<HashMap<NodeId, ServiceNodeHandle>>,
+}
+
+impl Simulator for ServiceSim {
+    fn new(_rand: &GlobalRng, _time: &TimeHandle, _config: &Config) -> Self
+    where
+        Self: Sized,
+    {
+        Self::default()
+    }
+    fn create_node(&self, id: NodeId) {
+        let mut handles = self.handles.lock().unwrap();
+        handles.insert(id, ServiceNodeHandle::default());
+    }
+    fn reset_node(&self, _id: NodeId) {}
```

Comment:
Don't forget to implement this function. Perhaps inserting a new `ServiceNodeHandle` to replace the old one.